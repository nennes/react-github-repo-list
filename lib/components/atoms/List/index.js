/* @flow */

import React from "react";

type Props = {
  children: Array<React.Element<*>>
};

const List = (props: Props) => {
  return (
    <ul className="list">
      {props.children.map((el, index) => {
        return (
          <li key={index}>
            {el}
          </li>
        );
      })}
    </ul>
  );
};

export default List;
