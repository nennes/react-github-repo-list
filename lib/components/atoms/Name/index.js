/* @flow */

import React from "react";
import { Repo as RepoType } from "../../../types";

const Name = ({ name, url }: RepoType) => {
  return (
    <div className="repo__name">
      <a href={url} target="_blank">
        <span title={name}>
          {name}
        </span>
      </a>
    </div>
  );
};

export default Name;
