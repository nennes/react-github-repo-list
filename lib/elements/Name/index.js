/* @flow */

import React from "react";

const Name = ({ name, url }: { name: string, url: string }) => {
  return (
    <div className="repo__name">
      <a href={url} target="_blank">
        <span title={name}>
          {name}
        </span>
      </a>
    </div>
  );
};

export default Name;
