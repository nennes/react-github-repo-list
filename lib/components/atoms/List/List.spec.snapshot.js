/* global test, expect */

import React from "react";
import List from "./index";
import Name from "../Name";
import renderer from "react-test-renderer";

test("List atom renders correctly", () => {
  const props = {
    children: [
      <div key="div">A child</div>,
      <Name key="comp" name="me" url="http://www.test.com" />
    ]
  };

  const component = renderer.create(<List {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
