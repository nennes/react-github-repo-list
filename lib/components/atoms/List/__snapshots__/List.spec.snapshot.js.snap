// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List atom renders correctly 1`] = `
<ul
  className="list"
>
  <li>
    <div>
      A child
    </div>
  </li>
  <li>
    <div
      className="repo__name"
    >
      <a
        href="http://www.test.com"
        target="_blank"
      >
        <span
          title="me"
        >
          me
        </span>
      </a>
    </div>
  </li>
</ul>
`;
